lib/spatial_features/validation.rb

```diff
@@ -0,0 +1,42 @@
+module SpatialFeatures
+  module Validation
+    # SHP file must come first
+    REQUIRED_SHAPEFILE_COMPONENT_EXTENSIONS = %w[shp shx dbf prj].freeze
+
+    # Check if a shapefile archive includes the required component files, otherwise
+    # raise an exception.
+    #
+    # @param [String] file_path                   Path of the zip archive
+    # @param [Zip::File] zip_file                 A Zip::File object
+    # @param [String] default_proj4_projection    Optional, if supplied we don't raise an exception when we're missing a .PRJ file
+    # @param [Boolean] allow_generic_zip_files    When true, we skip validation entirely if the archive does not contain a .SHP file
+    def self.validate_shapefile_archive!(archive_path, zip_file, default_proj4_projection: nil, allow_generic_zip_files: false)
```

Comment:
I'm a bit confused about why we need `archive_path`. Isn't having all the zip_file entries enough? Also, I think we need to validate that all the components have the same `basename`. For example, I think a shapefile is still invalid if it has all the required extensions but the files are `foo.shp`, `bar.shx`, `baz.prj`, etc.

Reply:
I piggy-backed blindly off some existing code I found doing that `basename` stuff without actually looking at the spec.

I did a little bit of a read but didn't see anything specific about filenames (https://www.esri.com/content/dam/esrisites/sitecore-archive/Files/Pdfs/library/whitepapers/pdfs/shapefile.pdf).  What you said about the basename being consistent makes sense.  

In the end, we're not trying to create a perfect/strict validator, so it's kind of up to us to choose a heuristic.  

Do you have any other sources I should look at to see the actual spec, or should we just ...

- ignore zip file name AND
- ensure all four of the required files have the same basename

?

Reply:
I think a heuristic that matches what we expect elsewhere in the gem makes sense. I believe that `rgeo-shapefile` expects all the basenames to be the same (https://github.com/culturecode/rgeo-shapefile/blob/master/lib/rgeo/shapefile/reader.rb#L179-L183), so let's go with that.

Reply:
Validating against the include SHP file's basename is wired up now.